File: python/64.py
class Solution:
    def sumNums(self, n: int) -> int:
        """
        使用and运算符短路
        """
        sum_val = 0       
        def _sum(n):
            nonlocal sum_val
            sum_val += n
            return n and _sum(n-1)
        _sum(n)
        return sum_val

File: cpp/64.cpp
class Solution {
public:
    int sumNums(int n) {
        int sum = 0;
        _sum(n, sum);
        return sum;
    }
    int _sum(int n, int &sum) {
        sum += n;
        return n && _sum(n-1, sum);
    }
};